### ISSUE TITLE
[ycql] Client authentication always queries the master

### ISSUE THREAD

**Issue by danielmh0** [yugabyte/yugabyte-db#6400]
YCQL authentication seems to always trigger a master query when a new client connection is opened.

Here's a snippet to identify the query:
```
  client: YQL_CLIENT_CQL
  request_id: 7135672164663255918                                                                                                                                                   schema_version: 0
  hash_code: 55498                                                                                                                                                                  hashed_column_values {                                                                                                                                                              value {
      string_value: "cassandra"                                                                                                                                                       }                                                                                                                                                                               }
  selected_exprs {
    column_id: 4
  }
  selected_exprs {
    column_id: 1
  }
  distinct: false
  max_hash_code: 55498
  column_refs {
    ids: 0
    ids: 1
    ids: 4
  }
  query_id: 1229201920
  rsrow_desc {
    rscol_descs {
      name: "salted_hash"
      ql_type {
        main: STRING
      }
    }
    rscol_descs {
      name: "can_login"
      ql_type {
        main: BOOL
      }
    }
  }
  is_forward_scan: true
  is_aggregate: false
}
include_trace: false
consistency_level: STRONG
```

This is another manifestation of https://github.com/yugabyte/yugabyte-db/issues/4640, similar to https://github.com/yugabyte/yugabyte-db/issues/5872.

Can we either cache or ideally eliminate this master query?

**Comment by kmuthukk** [comment]
Closing this as a duplicate of #6010 .

**Comment by ddorian** [comment]
Closed as duplicate of #6010
